games-board/stockfish/stockfish-sf_8.recipe

```diff
@@ -0,0 +1,72 @@
+SUMMARY="A chess engine claimed to be strongest in the world"
+DESCRIPTION="Stockfish is a free UCI chess engine derived from Glaurung 2.1. It \
+is not a complete chess program and requires some UCI-compatible GUI (e.g. \
+XBoard with PolyGlot, eboard, Arena, Sigma Chess, Shredder, ChessPartner or \
+Fritz) in order to be used comfortably. Read the documentation for your GUI of \
+choice for information about how to use Stockfish with it."
+HOMEPAGE="https://www.stockfishchess.org/"
+COPYRIGHT="2004-2008 Tord Romstad (Glaurung author)
+	2008-2017 Marco Costalba, Joona Kiiski, Gary Linscott, Tord Romstad"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/official-stockfish/Stockfish/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="e2b77d801d8837c22c2391b8cae08622c1434a2bb5947ddb08b081e1ad35c76c"
+SOURCE_DIR="Stockfish-sf_8"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="!x86_gcc2 x86"
+
+if [ "$targetArchitecture" != x86_gcc2 ]; then
+	if [ "$targetArchitecture" == arm ]; then
+		commandBinDir=$binDir
+		commandArch=armv7
+	elif [ "$targetArchitecture" == ppc ]; then
+		commandBinDir=$binDir
+		commandArch=ppc-32
+	elif [ "$targetArchitecture" == x86 ]; then
+		commandBinDir=$prefix/bin
+		commandArch=x86-32
+	elif [ "$targetArchitecture" == x86_64 ]; then
+		commandBinDir=$binDir
+		commandArch=x86-64
+	fi
+else
+	commandBinDir=$prefix/bin
+	commandArch=x86-32
+fi
```

Comment:
The rule is that we set commandBinDir to $binDir for all primary architectures where it builds fine.

If it does not build for x86_gcc2, then, for that arch only, we set commandBinDir to $prefixBin so that we can build stockfish_x86, that is, we build it on x86_gcc2 but using the x86 secondary arch.
The result of the commandBinDir trick is that instead of installing the runtimes in /system/bin/x86/ we install then in /system/bin/, where you know we won't ever clash with the runtimes built for x86_gcc2 primary arch since we know it does not build on x86_gcc2 primary arch.